### ISSUE TITLE
Uncaught TypeError: Cannot read property 'children' of undefined

### ISSUE THREAD

**Issue by alex-martin-xactium** [joeferraro/MavensMate-Atom#149]
[Enter steps to reproduce below:]

1. Went to Project Edit
2. After metadata refresh, this appeared.

**Atom Version**: 0.179.0
**System**: Mac OS X 10.10.1
**Thrown From**: [MavensMate-Atom](https://github.com/joeferraro/MavensMate-Atom) package, v0.0.18

### Stack Trace

Uncaught TypeError: Cannot read property 'children' of undefined

```
At /Users/alexmartin/.atom/packages/MavensMate-Atom/node_modules/mavensmate/lib/mavensmate/project.js:848

TypeError: Cannot read property 'children' of undefined
  at /Users/alexmartin/.atom/packages/MavensMate-Atom/node_modules/mavensmate/lib/mavensmate/project.js:848:57
  at Function.forEach (/Applications/Atom.app/Contents/Resources/app/node_modules/roaster/node_modules/cheerio/node_modules/lodash/dist/lodash.js:3297:15)
  at /Users/alexmartin/.atom/packages/MavensMate-Atom/node_modules/mavensmate/lib/mavensmate/project.js:838:19
  at Function.forOwn (/Applications/Atom.app/Contents/Resources/app/node_modules/roaster/node_modules/cheerio/node_modules/lodash/dist/lodash.js:2105:15)
  at /Users/alexmartin/.atom/packages/MavensMate-Atom/node_modules/mavensmate/lib/mavensmate/project.js:824:15
  at /Users/alexmartin/.atom/packages/MavensMate-Atom/node_modules/mavensmate/node_modules/fs-extra/node_modules/jsonfile/lib/jsonfile.js:23:5
  at fs.js:337:14
  at FSReqWrap.oncomplete (fs.js:100:15)

```

### Commands

```
     -0:17.8 mavensmate:edit-project (atom-text-editor.editor)
```

### Config

```json
{
  "core": {
    "themes": [
      "atom-dark-ui",
      "monokai"
    ]
  }
}
```

### Installed Packages

```coffee
# User
MavensMate-Atom, v0.0.18
file-icons, v1.4.11
git-control, v0.1.8
git-log, v0.2.0
git-projects, v1.7.1
highlight-selected, v0.9.0
minimap, v4.2.2
monokai, v0.12.0
project-manager, v1.15.5

# Dev
No dev packages
```

**Comment by joeferraro** [comment]
@alex-martin-xactium can you tell me which objects were in your subscription when this issue occurred?

**Comment by joeferraro** [comment]
Can you please post your package.xml to this issue? (you can obfuscate/change sensitive names if you need to)

**Comment by alex-martin-xactium** [comment]
Company would prefer I don't post the package.xml.

For more information, the project I was using was created in Sublime MM and worked fine there. I also created a new dev org, add some basic files and changed the package.xml to be in a similar format as the other and the error wouldn't appear (package.xml was using explicit names instead of *)

Not sure how helpful this is, but this is the stripped down version of the xml.

```xml
<?xml version="1.0" encoding="UTF-8"?>
<Package xmlns="http://soap.sforce.com/2006/04/metadata">
	<types>
		<name>ApexClass</name>
	</types>
	<types>
		<name>ApexComponent</name>
	</types>
	<types>
		<name>ListView</name>
	</types>
	<types>
		<name>ApexPage</name>
	</types>
	<types>
		<name>ApexTrigger</name>
	</types>
	<types>
		<name>CustomApplication</name>
	</types>
	<types>
		<name>CustomField</name>
	</types>
	<types>
		<members>CustomLabels</members>
		<name>CustomLabels</name>
	</types>
	<types>
		<name>CustomObject</name>
	</types>
	<types>
		<name>CustomPageWebLink</name>
	</types>
	<types>
		<name>Dashboard</name>
	</types>
	<types>
		<name>Document</name>
	</types>
	<types>
		<name>EmailTemplate</name>
	</types>
	<types>
		<name>Layout</name>
	</types>
	<types>
		<members>Risk_Letterhead</members>
		<name>Letterhead</name>
	</types>
	<types>
		<members>Admin</members>
		<members>ReadOnly</members>
		<name>Profile</name>
	</types>
	<types>
		<members>InternationalEscalations</members>
		<members>InternationalLeads</members>
		<members>InternationalPlatinumGold</members>
		<members>InternationalSilverBronze</members>
		<members>PartnerRelations</members>
		<members>USEscalations</members>
		<members>USLeads</members>
		<members>USPlatinumGold</members>
		<members>USSilverBronze</members>
		<name>Queue</name>
	</types>
	<types>
		<members>Idea.InternalIdeasIdeaRecordType</members>
		<name>RecordType</name>
	</types>
	<types>
		<members>unfiled$public/SampleReportMonthtoDateTrending</members>
		<members>unfiled$public/SampleReportOpenCasesByPriority</members>
		<members>unfiled$public/SampleReportOpenDeals</members>
		<members>unfiled$public/SampleReportPipelineByCloseMonth</members>
		<members>unfiled$public/SampleReportPipelineHistory</members>
		<members>unfiled$public/SampleReportSalesLeaderboard</members>
		<members>unfiled$public/SampleReportSalesPipelinebyStage</members>
		<members>unfiled$public/SampleReportSolutionsAdded</members>
		<members>unfiled$public/SampleReportTopAccounts</members>
		<members>unfiled$public/SampleReportUsersLoggedIn</members>
		<members>unfiled$public/SampleReportofAccounts</members>
		<members>unfiled$public/SampleReportofCases</members>
		<members>unfiled$public/SampleReportofContacts</members>
		<members>unfiled$public/SampleReportofDocuments</members>
		<members>unfiled$public/SampleReportofLeads</members>
		<members>unfiled$public/SampleReportofOpenTasks</members>
		<members>unfiled$public/SampleReportofOpportunities</members>
		<members>unfiled$public/SampleReportofReports</members>
		<members>unfiled$public/SampleReportofSolutions</members>
		<members>unfiled$public/SampleReportofTasksandActivities</members>
		<members>unfiled$public/SampleReportsLeadsbyStatus</members>
		<name>Report</name>
	</types>
	<types>
		<name>ReportType</name>
	</types>
	<types>
		<members>AboutBanner</members>
		<members>CanvasJs</members>
		<members>CanvasJs</members>
		<members>DrawHeatmap</members>
		<members>HelpLogo</members>
		<members>IUI</members>
		<members>ManageLinks_JS</members>
		<members>ReviewResultCollectionStyles</members>
		<members>RiskClimateDownGreenArrow</members>
		<members>RiskClimateRightYellowArrow</members>
		<members>RiskClimateUpRedArrow</members>
		<members>RiskManagerStyles</members>
		<members>SiteSamples</members>
		<members>canvasObjectHandler</members>
		<members>empResidualDirection</members>
		<members>heatmap_circle</members>
		<members>hmc_000000</members>
		<members>hmc_2E2EFE</members>
		<members>hmc_2E9AFE</members>
		<members>hmc_41FBAA</members>
		<members>hmc_59E817</members>
		<members>hmc_68EFEA</members>
		<members>hmc_A1FB41</members>
		<members>hmc_F62817</members>
		<members>hmc_F7FE2E</members>
		<members>hmc_FD3EB7</members>
		<members>hmc_FDD017</members>
		<members>hmc_black</members>
		<members>hmc_gray</members>
		<members>icons</members>
		<members>riskAssessmentEntry</members>
		<members>swapImage</members>
		<name>StaticResource</name>
	</types>
	<types>
		<members>Account</members>
		<members>AppetiteReportSetting__c</members>
		<members>Asset</members>
		<name>Workflow</name>
	</types>
	<types>
		<members>About</members>
		<members>Cause__c</members>
		<name>CustomTab</name>
	</types>
	<types>
		<members>Account.AltavistaNews</members>
		<members>Account.Billing</members>
		<members>Account.GoogleMaps</members>
		<members>Account.GoogleNews</members>
		<members>Account.GoogleSearch</members>
		<members>Account.HooversProfile</members>
		<members>Account.YahooMaps</members>
		<members>Campaign.ViewAllCampaignMembers</members>
		<members>Campaign.ViewCampaignInfluenceReport</members>
		<members>Case.UpsellCrosssellOpportunity</members>
		<members>Contact.GoogleMaps</members>
		<members>Contact.GoogleSearch</members>
		<members>Contact.SendGmail</members>
		<members>Contact.YahooMaps</members>
		<members>Contact.YahooWeather</members>
		<members>Lead.AltavistaNews</members>
		<members>Lead.GoogleMaps</members>
		<members>Lead.GoogleNews</members>
		<members>Lead.GoogleSearch</members>
		<members>Lead.HooversProfile</members>
		<members>Lead.SendGmail</members>
		<members>Lead.YahooMaps</members>
		<members>Opportunity.DeliveryStatus</members>
		<name>WebLink</name>
	</types>
	<version>31.0</version>
</Package>
```

All the files were referenced explicitly, instead of using *.

**Comment by alex-martin-xactium** [comment]
This error has reared it's ugly head again. Any ideas?

**Comment by GaryBurgin** [comment]
Hi, I seem to have run into the same issue.
It appears to be due to the project.js attempting to match against the Project Metadata Index (.org_metadata). If a reference in the src package.xml doesn't exist in the .org_metadata it throws an exception when trying to access the children array (of undefined).
I imagine you could use the MavensMate > Metadata > Update Project Metadata Index (.org_metadata) to refresh the .org_metadata if you are able to initialise your MavensMate project. In my case the exception stops the initialisation, as a result I cannot do this so assume I would need to correct this manually or remove unreferenced files from my src package.xml.

Please note the above comments are based on a 2 min investigation into this issue, there could be further issues, but this may provide a starting point to resolve this issue.

Thank you.

**Comment by GaryBurgin** [comment]
I was naughty and stuck a try catch here to get me past the project initialisation.
This worked and I am now able to continue, as reference for @joeferraro please see below, I know this is not the best fix, just a work around, but may help you in your update to resolve this issue for others.

temp fix added to mavensmate > lib > mavensmate > project.js

![screen shot 2015-03-09 at 15 45 10](https://cloud.githubusercontent.com/assets/1515693/6558361/b1e06de4-c673-11e4-8b65-345f637830e4.png)

**Comment by abbas133** [comment]
@joeferraro I am also facing the same issue, but thanks to @GaryBurgin, this try-catch block patch is working for now.

**Comment by nwoolls** [comment]
Noting that I also ran into this issue trying the Atom package - editing project.js as shown above worked around the issue.

**Comment by joeferraro** [comment]
fyi this is resolved in the upcoming 1.0.0

**Comment by abbas133** [comment]
Thanks @joeferraro
